system/jlib/jptree.cpp

```diff
@@ -3176,6 +2818,408 @@ bool PTree::checkPattern(const char *&xxpath) const
     return ret;
 }
 
+//////////////////////
+
+unsigned ChildMapC::getHashFromElement(const void *e) const
+{
+    LocalPTree &elem = (LocalPTree &) (*(IPropertyTree *)e);
+    const char *name = elem.queryName();
+    return hashc((const unsigned char *)name, (size32_t)strlen(name), 0);
```

Comment:
It looks like I lost a comment.

Have you considered the efficiency impact of repeating this hash?  Particularly when hash tables are expanded the cost can be quite high.

Simpler would be to always create an "atom", but only common it up in dali.  It would avoid creating AtomChildMap etc..